src/org/zaproxy/zap/extension/callback/ExtensionCallback.java

```diff
@@ -98,12 +98,14 @@ private void restartServer(int port) {
 
     public String getCallbackAddress() {
         String addr = this.getCallbackParam().getRemoteAddress();
+        boolean isSecure = this.getCallbackParam().isSecure();
+        String schema = isSecure ? "https" : "http";
         if (addr.contains(":")) {
             // Looks like its IPv6
-            return "http://[" + addr + "]:" + actualPort + "/";
+            return schema + "://[" + addr + "]:" + actualPort + "/";
```

Comment:
Worth remove the code duplication: `addr = "[" + addr + "]";` (and drop the IPv4 comment).

Reply:
You mean this way?

```     
boolean ipv6 = addr.contains(":");
String ipAddr = ipv6 ? "[" + addr + "]" : addr;   
return schema + "://" + ipAddr + ":" + actualPort + "/";
```

Reply:
Was suggesting:
```Java
if (addr.contains(":")) {
    // Looks like its IPv6
    addr = "[" + addr + "]";
}
return scheme + "://" + addr + ":" + actualPort + "/";
```
but that's fine as well (although the variable `ipAddr` should have other name, `addr` might be a hostname).